app/src/hooks/useMarkets.tsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react'
+
+import { ConnectedWeb3Context } from './connectedWeb3'
+import { useContracts } from './useContracts'
+import { getLogger } from '../util/logger'
+import { Market, MarketAndQuestion, MarketStatus, Status } from '../util/types'
+
+const logger = getLogger('Market::useMarkets')
+
+export const useMarkets = (
+  context: ConnectedWeb3Context,
+): {
+  markets: MarketAndQuestion[]
+  status: Status
+} => {
+  const { marketMakerFactory, conditionalTokens, realitio } = useContracts(context)
+
+  const [status, setStatus] = useState<Status>(Status.Ready)
+  const [markets, setMarkets] = useState<MarketAndQuestion[]>([])
+
+  useEffect(() => {
+    const fetchMarkets = async () => {
+      try {
+        setStatus(Status.Loading)
+        const provider = context.library
+
+        const getQuestionData = async (market: Market) => {
+          const { conditionId } = market
+          // Get question data
+          const questionId = await conditionalTokens.getQuestionId(conditionId, provider)
```

Comment:
Let's refactor `getQuestionId` so that it doesn't receive the provider as an argument (the instance already has access to the provider).

Reply:
Good catch, done!